core/metamodel.class.php

```diff
@@ -2889,6 +2889,7 @@ public static function InitClasses($sTablePrefix)
 					if (!empty($aCurrentClassUniquenessRules))
 					{
 						$aClassFields = self::GetAttributesList($sPHPClass);
+						$aClassFields[] = "finalclass";
```

Comment:
`finalclass` field name can be customized (`classes/class/properties/db_final_class_field` node in the XML)

I suggest using this instead of a hardcoded string :

```suggestion
						$aClassFields[] = self::DBGetClassField($sPHPClass);
```

But anyway the uniqueness rule check is done in the wrong place as magix fields are not already added to classes...